atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -734,4 +787,13 @@ private void testValidateExternalAndCommitLocks(final LockAwareTransactionTask<V
                 lockAwareTransactionTask);
     }
 
+    /**
+     * Hack to get reference to underlying {@link SnapshotTransaction}. See how transaction managers are composed at
+     * {@link AtlasDbTestCase#setUp()}.
+     */
+    private static SnapshotTransaction unwrapSnapshotTransaction(Transaction cachingTransaction) {
```

Comment:
I'm not terribly proud of this, and can take it back out if we think it's too brittle.